File: src/main/java/com/fish/entity/pojo/TopicSolutions.java
package com.fish.entity.pojo;

import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 *  实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(value = "oj_topic_solutions")
public class TopicSolutions implements Serializable {
    @Id(keyType = KeyType.Auto)
    private Long solutionId;
    private Long topicId;
    private String title;
    private String content;
    /**
     * 置顶
     */
    private Boolean pined;
    private Boolean enabled;
    private LocalDateTime createTime;

}

File: src/main/kotlin/com/fish/entity/pojo/TopicSolutions.kt
package com.fish.entity.pojo

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import com.mybatisflex.annotation.Id
import com.mybatisflex.annotation.KeyType
import com.mybatisflex.annotation.Table
import java.io.Serializable
import java.time.LocalDateTime

/**
 * 实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Table(value = "oj_topic_solutions")
data class TopicSolutions(
    @Id(keyType = KeyType.Auto) var solutionId: Long?,
    var topicId: Long?,
    var title: String?,
    var content: String?,

    /**
     * 置顶
     */
    var pined: Boolean?,
    var enabled: Boolean?,

    @JsonSerialize(using = LocalDateTimeSerializer::class)
    @JsonDeserialize(using = LocalDateTimeDeserializer::class)
    var createTime: LocalDateTime?,
) : Serializable
